### PR TITLE
release 0.13.1

### PR DESCRIPTION
**Pull request by Shishir435**
<h3>Greptile Summary</h3>

This release updates the extension to 0.13.1 and adds fixes across provider discovery, permission recovery, chat persistence, embeddings, PDF processing, Markdown rendering, tab extraction, and model lifecycle.

- Adds Ollama Cloud discovery, metadata, UI badges, documentation, and localized copy.
- Persists permission-blocked turn context and resumes generation after access is granted.
- Improves provider catalog normalization, embedding cache/index behavior, session-tree recovery, and SQLite projections.
- Consolidates tab extraction refresh, throttles streaming Markdown parsing, and hardens loaded-model refresh lifecycle.

<h3>Confidence Score: 4/5</h3>

The PR appears safe to merge, with a non-blocking message-tree fallback edge case worth correcting for imported histories.

Most changed flows preserve their existing contracts and add focused tests, but missing session leaves are selected by timestamp rather than structural leaf status, which can misattach later messages in imported non-monotonic trees.

**Files Needing Attention:** src/features/sessions/stores/chat-session-message-actions.ts

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src/features/chat/lib/resume-permission-turn.ts | Coordinates permission requests, branch navigation, stream admission, resumed generation, and notice resolution. |
| src/features/sessions/stores/chat-session-message-actions.ts | Adds missing-leaf recovery, but timestamp-based fallback can select an internal tree node for imported non-monotonic histories. |
| src/lib/providers/ollama.ts | Supplements local Ollama models with bounded, non-fatal cloud recommendations and normalizes cloud metadata. |
| src/lib/providers/openai-compatible.ts | Broadens model-catalog parsing while retaining valid entries when unrelated metadata is malformed. |
| src/lib/embeddings/search.ts | Changes hybrid fusion to rehydrate surviving candidates from persistent vector storage. |
| src/background/tab-lifecycle.ts | Registers tab-removal cleanup for the bounded tab-content cache. |
| src/application/file-processing/processors/pdf-processor.ts | Adds timeout-bounded worker loading with a no-worker fallback for PDF extraction. |
| packages/contracts/src/chat-activity.ts | Defines the versioned permission-resume snapshot and resolved-notice contract. |

</details>

<details><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart LR
  U[User submits message] --> P{Optional permission available?}
  P -- No --> N[Persist permission notice and resume snapshot]
  N --> G[User grants permission]
  G --> R[Rebuild durable turn from snapshot]
  P -- Yes --> R
  R --> C[Build context]
  C --> M[Stream from selected provider/model]
  M --> D[Persist assistant response]
```
</details>

<a href="https://app.greptile.com/ide/claude-code?prompt=Greploop%20shishir435%2Follama-client%20PR%20%23305%3A%20work%20through%20Greptile's%20open%20review%20comments%2C%20then%20keep%20reviewing%20and%20fixing%20until%20it%20comes%20back%20clean%20at%205%2F5%20with%20zero%20unresolved%20comments.%0AStart%20by%20reading%20the%20comments%20off%20the%20PR%20itself.%20On%20GitHub%2C%20use%20paginated%20%60gh%20api%20graphql%60%20to%20query%20%60pullRequest.reviewThreads%60%20with%20each%20thread's%20%60isResolved%60%20value%20and%20inline%20%60comments%60.%20%60gh%20pr%20view%20--comments%60%20only%20includes%20conversation%20comments%2C%20so%20do%20not%20use%20it%20as%20the%20findings%20source.%20They%20are%20not%20listed%20in%20this%20prompt%20on%20purpose%3A%20the%20PR%20is%20current%2C%20a%20pasted%20copy%20would%20not%20be.%20Skip%20anything%20already%20resolved%2C%20and%20if%20you%20judge%20a%20comment%20wrong%2C%20say%20so%20rather%20than%20changing%20code%20to%20satisfy%20it.%0A%0APrefer%20the%20Greptile%20CLI%2C%20which%20reviews%20the%20working%20tree%20with%20no%20push%20and%20no%20CI%20run.%20Fall%20back%20to%20PUSH%20LOOP%20only%20where%20a%20step%20below%20says%20to.%0A1.%20Run%20%60command%20-v%20greptile%60.%20Missing%3A%20go%20to%20PUSH%20LOOP%2C%20and%20pitch%20the%20CLI%20at%20the%20end.%0A2.%20Run%20%60greptile%20whoami%60.%20It%20exits%200%20even%20when%20signed%20out%2C%20so%20read%20the%20output%3A%20if%20it%20says%20%22Not%20signed%20in%22%2C%20ask%20the%20user%20to%20run%20%60greptile%20login%60%20%28it%20opens%20a%20browser%2C%20do%20not%20run%20it%20yourself%29%20and%20go%20to%20PUSH%20LOOP%20with%20no%20pitch.%0A3.%20Look%20for%20%60greploop%2FSKILL.md%60%20under%20%60.agents%2Fskills%2F%60%20or%20%60.claude%2Fskills%2F%60%2C%20both%20in%20the%20repo%20root%20and%20in%20the%20user's%20home%20directory.%20Found%3A%20follow%20that%20skill%2C%20never%20pushing%20between%20rounds%2C%20and%20you%20are%20done.%0A4.%20Missing%3A%20run%20%60greptile%20skills%20list%60.%20If%20it%20fails%20as%20an%20unknown%20command%2C%20this%20CLI%20predates%20the%20skill%2C%20so%20tell%20the%20user%20to%20run%20%60greptile%20update%60%20and%20go%20to%20PUSH%20LOOP%20with%20no%20pitch.%20Otherwise%20ask%20the%20user%20before%20running%20%60greptile%20skills%20install%20greploop%60%2C%20then%20follow%20the%20skill%20it%20writes.%0A%0APUSH%20LOOP%20%28only%20when%20a%20step%20above%20sends%20you%20here%29%3A%0A%0AGREPLOOP%20ENABLED%3A%20After%20fixing%20all%20issues%2C%20commit%20and%20push%20your%20changes.%20Greptile%20will%20automatically%20re-review%20the%20PR%20via%20webhook.%20Wait%20for%20the%20review%20to%20complete%2C%20then%20read%20the%20new%20review%20comments%20and%20confidence%20score.%20If%20the%20confidence%20score%20is%20below%205%2F5%20or%20there%20are%20unresolved%20comments%2C%20fix%20those%20issues%20and%20push%20again.%20Repeat%20until%20the%20confidence%20score%20is%205%2F5%20with%20zero%20unresolved%20comments.%20Maximum%205%20iterations.%20Do%20NOT%20open%20a%20new%20PR%20%E2%80%94%20keep%20pushing%20to%20the%20same%20branch.%0A%0AIF%20YOU%20ENDED%20UP%20ON%20PUSH%20LOOP%20BECAUSE%20THE%20CLI%20WAS%20MISSING%3A%20once%20the%20loop%20is%20done%2C%20print%20the%20block%20below%20and%20nothing%20else%20about%20the%20CLI.%20Reproduce%20it%20character%20for%20character%20inside%20a%20fenced%20code%20block%2C%20so%20the%20borders%20line%20up.%20Do%20not%20reword%20it%2C%20translate%20it%2C%20summarize%20it%2C%20add%20to%20it%2C%20or%20drop%20the%20box.%20Print%20it%20once%2C%20and%20do%20not%20bring%20the%20CLI%20up%20again%20later%20in%20the%20session.%0A%0A%E2%95%AD%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%95%AE%0A%E2%94%82%20%20You%20can%20run%20greploops%20faster%20locally%20with%20our%20CLI.%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%E2%94%82%0A%E2%94%82%20%20Install%20it%20at%20https%3A%2F%2Fwww.greptile.com%2Fcli%2C%20or%20I%20can%20install%20it%20for%20you.%20%20%E2%94%82%0A%E2%95%B0%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%95%AF%0A%0AIf%20they%20take%20you%20up%20on%20it%2C%20install%20with%20%60npm%20install%20-g%20greptile%60%20%28or%20%60brew%20install%20greptileai%2Ftap%2Fgreptile%60%29%2C%20then%20%60greptile%20skills%20install%20greploop%60.%20Leave%20%60greptile%20login%60%20to%20them%2C%20it%20opens%20a%20browser.&repo=shishir435%2Follama-client&pr=305&platform=github">Fix all with Greploop</a>

<a href="https://app.greptile.com/ide/claude-code?prompt=%23%23%23%20Issue%201%0Asrc%2Ffeatures%2Fsessions%2Fstores%2Fchat-session-message-actions.ts%3A104%0A**Timestamp%20fallback%20can%20select%20branches**%0A%0AIf%20an%20imported%20session%20has%20no%20valid%20%60currentLeafId%60%20and%20preserves%20non-monotonic%20timestamps%2C%20this%20fallback%20selects%20the%20newest%20timestamped%20row%20even%20when%20it%20has%20children.%20The%20displayed%20path%20then%20omits%20its%20existing%20child%20branch%2C%20and%20the%20next%20message%20is%20persisted%20under%20that%20internal%20node%20instead%20of%20extending%20a%20structural%20leaf.%0A%0A---%0A%0AFor%20each%20issue%20above%2C%20determine%20whether%20it%20is%20valid%20and%20should%20be%20fixed.%20If%20so%2C%20fix%20it%20directly.&repo=shishir435%2Follama-client&pr=305&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaudeDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaude.svg?v=6">Fix All in Claude Code</picture></a> <a href="https://app.greptile.com/api/ide/codex?prompt=IMPORTANT%3A%20Work%20in%20the%20repository%20%22shishir435%2Follama-client%22%20on%20the%20existing%20branch%20%22preview%22.%20Checkout%20that%20branch%20%E2%80%94%20do%20NOT%20create%20a%20new%20branch%20or%20open%20a%20new%20PR.%20Push%20your%20changes%20to%20%22preview%22.%0A%0A%23%23%23%20Issue%201%0Asrc%2Ffeatures%2Fsessions%2Fstores%2Fchat-session-message-actions.ts%3A104%0A**Timestamp%20fallback%20can%20select%20branches**%0A%0AIf%20an%20imported%20session%20has%20no%20valid%20%60currentLeafId%60%20and%20preserves%20non-monotonic%20timestamps%2C%20this%20fallback%20selects%20the%20newest%20timestamped%20row%20even%20when%20it%20has%20children.%20The%20displayed%20path%20then%20omits%20its%20existing%20child%20branch%2C%20and%20the%20next%20message%20is%20persisted%20under%20that%20internal%20node%20instead%20of%20extending%20a%20structural%20leaf.%0A%0A---%0A%0AFor%20each%20issue%20above%2C%20determine%20whether%20it%20is%20valid%20and%20should%20be%20fixed.%20If%20so%2C%20fix%20it%20directly.&repo=shishir435%2Follama-client&pr=305&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodexDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodex.svg?v=6">Fix All in Codex</picture></a>

<sub>Reviews (1): Last reviewed commit: ["0.13.1: restore the store name and descr..."](https://github.com/shishir435/ollama-client/commit/42970b6f30a7ab870a7e2b896f445da28c1f9c79) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=53652948)</sub>

> Greptile also left **1 inline comment** on this PR.

<details><summary><h4>Context used (5)</h4></summary>

- Knowledge Base — [Background Runtime](https://app.greptile.com/shishir435/-/custom-context/knowledge-base/shishir435/ollama-client/-/docs/background-runtime.md)
- Knowledge Base — [Chat Flow: Send, Stream, Cancel, Edit, and Fork](https://app.greptile.com/shishir435/-/custom-context/knowledge-base/shishir435/ollama-client/-/docs/chat-flow.md)
- Knowledge Base — [Local RAG: File Upload, Chunking, Embedding, and Retrieval](https://app.greptile.com/shishir435/-/custom-context/knowledge-base/shishir435/ollama-client/-/docs/knowledge-rag.md)
- Knowledge Base — [OPFS Single-Owner SQLite Persistence](https://app.greptile.com/shishir435/-/custom-context/knowledge-base/shishir435/ollama-client/-/docs/persistence-sqlite.md)
- Knowledge Base — [Provider Abstraction Layer](https://app.greptile.com/shishir435/-/custom-context/knowledge-base/shishir435/ollama-client/-/docs/providers.md)
</details>

### CODE REVIEW

File: src/features/sessions/stores/chat-session-message-actions.ts [hunk 1]
```diff
@@ -59,23 +77,64 @@ export const createChatSessionMessageActions = (
...
+     * safe to hand `addMessage` as the next parent. The decode-failure case
+     * cannot arrive here at all — it returned above rather than guess.
+     *
+     * Still not written back: a read does not repair durable state.
+     */
+    const leafId = treeNodes.some(
+      (node) => String(node.id) === String(storedLeafId)
+    )
+      ? storedLeafId
+      : treeNodes[treeNodes.length - 1].id
```
[the remark this answers is not in the document]
Comment (Shishir435) [reply src/features/sessions/stores/chat-session-message-actions.ts:104]: fix it  in 0.13.2
